ports/servoshell/minibrowser.rs

```diff
@@ -78,62 +130,127 @@ impl Minibrowser {
+                                        event_queue.borrow_mut().push(MinibrowserEvent::Go);
+                                        location_dirty.set(false);
+                                    }
 
-                                let location_field = ui.add_sized(
-                                    ui.available_size(),
-                                    egui::TextEdit::singleline(&mut *location.borrow_mut()),
-                                );
+                                    let location_field = ui.add_sized(
+                                        ui.available_size(),
+                                        egui::TextEdit::singleline(&mut *location.borrow_mut()),
+                                    );
 
-                                if location_field.changed() {
-                                    location_dirty.set(true);
-                                }
-                                if ui.input(|i| i.clone().consume_key(Modifiers::COMMAND, Key::L)) {
-                                    location_field.request_focus();
-                                }
-                                if location_field.lost_focus() &&
-                                    ui.input(|i| i.clone().key_pressed(Key::Enter))
-                                {
-                                    event_queue.borrow_mut().push(MinibrowserEvent::Go);
-                                    location_dirty.set(false);
+                                    if location_field.changed() {
+                                        location_dirty.set(true);
+                                    }
+                                    if ui.input(|i| {
+                                        i.clone().consume_key(Modifiers::COMMAND, Key::L)
+                                    }) {
+                                        location_field.request_focus();
+                                    }
+                                    if location_field.lost_focus() &&
+                                        ui.input(|i| i.clone().key_pressed(Key::Enter))
+                                    {
+                                        event_queue.borrow_mut().push(MinibrowserEvent::Go);
+                                        location_dirty.set(false);
+                                    }
+                                },
+                            );
+                        },
+                    );
+                    ui.cursor().min.y
+                });
+            *toolbar_height = Length::new(height);
+
+            CentralPanel::default()
+                .frame(Frame::none())
+                .show(ctx, |ui| {
+                    let min = ui.cursor().min;
+                    let size = ui.available_size();
+                    let rect = egui::Rect::from_min_size(min, size);
+                    ui.allocate_space(size);
+
+                    if let Some(servo_fbo) = servo_framebuffer_id {
```

Comment:
It think an early return might help reduce some of the nesting here:

```rust
let servo_fbo = match servo_framebuffer_id {
    Some(fbo) => fbo,
    None => return,
};
```

Reply:
done in a7d0b2e1f3dcfdfc68c222895b671c334f311c86